File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    counta = 0
    countb = 0
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    
    for i in s :
        print(i)
        if i == '(' :
            counta += 1
        elif i == ')' :
            countb += 1
        if countb > counta :
            answer = False
            break
    print(counta,countb)
    if counta == countb :
        answer = True
    else :
        answer = False

    if s[0] == ')' :
        answer = False
    elif s[len(s)-1] == '(' :
        answer = False
        


    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;

        int pointa = 1;
        int pointb = 1;
        
        for (int i =0; i<s.length(); i++){
            
            if(s.charAt(i) =='(') {
                pointa ++;
            }else{
                pointb ++;
            }
            
            if(pointb > pointa){
                answer = false;
                break;
            }
            
        }
        
        if(answer == true && pointa != pointb) {
            answer = false;
        }
        
        return answer;
    }
}